sunpy/map/header_helper.py

```diff
@@ -30,12 +30,14 @@ def meta_keywords():
     return _map_meta_keywords
 
 
-@u.quantity_input
-def make_fitswcs_header(data, coordinate, reference_pixel: u.pix = None,
-                        scale: u.arcsec/u.pix = None,
-                        rotation_angle: u.deg = None,
-                        rotation_matrix=None,
-                        **kwargs):
+@u.quantity_input(reference_pixel=u.pix, scale=u.arcsec/u.pix,
```

Comment:
The unit arguments to `quantity_input` should be function annotations (as it was before) only the equivalencies should be in the actual decorator call.

Reply:
so do you mean to put it back to the way it was in `make_fitswcs_header` such that `scale: u.arcsec/u.pix = None` etc, and only have `@u.quantity_input(wavelength=u.angstrom, equivalencies=u.spectral())` ?

Reply:
yes, but only have `@u.quantity_input(equivalencies=u.spectral())` and all the others as function annotations.

For what it's worth the equivalency will apply to all inputs. I don't think it matters though as the rest are not spectral units.

Reply:
as I see